cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -193,7 +196,11 @@ func (ns *nodeSyncer) sync(node string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to retrieve GSA list for Node %q: %v", node, err)
 	}
-	return ns.hms.sync(node, ns.location, gsaList)
+	zone, err := ns.zones.zoneByNode(node)
+	if err != nil {
+		return fmt.Errorf("failed to find zone for node %q: %w", node, err)
+	}
+	return ns.hms.sync(node, zone, gsaList)
```

Comment:
Sorry I had been a while ... can you please remind me about zone vs location wrt the HMS APIs?

Reply:
The us-west1 region has three zones: us-west1-a, us-west1-b, and us-west1-c (https://cloud.google.com/compute/docs/regions-zones) . The location here is the cluster's location. It can be a zone in a zonal cluster or a region in a regional cluster. But HMS API doesn't support region.